core/integration/module_typescript_test.go

```diff
@@ -1779,3 +1779,37 @@ class Test {
 		require.JSONEq(t, `{"test": {"foo": "bar"}}`, out)
 	})
 }
+
+func (TypescriptSuite) TestPrivatePropertyWithUnsupportedType(ctx context.Context, t *testctx.T) {
+	c := connect(ctx, t)
+
+	modGen := c.Container().From(golangImage).
+		WithMountedFile(testCLIBinPath, daggerCliFile(t, c)).
+		WithWorkdir("/work").
+		With(daggerExec("init", "--name=test", "--sdk=typescript")).
+		With(sdkSource("typescript", `import { object, func } from '@dagger.io/dagger'
+
+@object()
+export class Test {
+	@func()
+	foo: string
+
+	_bar: any
+
+	constructor() {
+		this.foo = 'foo'
+		this._bar = 42
+	}
+
+	@func()
+	bar(): string {
```

Comment:
What about if this type is something garbage and unresolvable? Like `thisdoesnotexist`?

Reply:
The return type (here: `string`) is exposed  so it has to be valid and supported by Dagger.
For `_bar`, it needs to be a valid typescript type (not necessarily supported by Dagger though) otherwise the runtime will just fail to run the code, like in go if you put `func a(b: thisdoesnotexist)`, the code will simply not compile.

Reply:
Fair!